src/collections/map.rs

```diff
@@ -188,6 +213,30 @@ where
         Ok(())
     }
 }
+/// An immutable owned reference to an existing value in a collection.
```

Comment:
Also this comment isn't quite accurate, maybe should be `A wrapper which only allows immutable access to its inner value.`.